source/software/module_system_basics.rst

```diff
@@ -0,0 +1,327 @@
+   Currently Loaded Modulefiles:
+     1) cluster/wice/batch
+     2) GCCcore/10.3.0
+     ...
+     16) OpenMPI/4.1.1-GCC-10.3.0
+     17) OpenBLAS/0.3.15-GCC-10.3.0
+     ...
+     46) PLUMED/2.9.0-foss-2023a
+     47) CP2K/2023.1-foss-2023a
+     48) GROMACS/2023.3-foss-2023a-PLUMED-2.9.0
+
+Note that this does not just show the two requested modules, but also all
+the modules that got loaded automatically in order to satisfy (runtime)
+dependencies of the explicitly loaded ``CP2K`` and ``GROMACS`` installations
+(``PLUMED``, ``OpenMPI``, ``OpenBLAS``, etcetera).
+
+
+Unloading modules
+~~~~~~~~~~~~~~~~~
+
+To unload a module, one can use the ``module unload`` command. It works
+consistently with the ``load`` command, and reverses the latter's
+effect. One can however unload automatically loaded modules manually, to
+debug some problem.
+
+::
+
+   $ module unload CP2K
+
+Notice that the version was not specified: the module system is
+sufficiently clever to figure out what the user intends. However,
+checking the list of currently loaded modules is always a good idea,
+just to make sure...
+
+
+.. _module_purge:
+
+Purging modules
+~~~~~~~~~~~~~~~
+
+In order to unload all modules at once and start with a clean slate, use:
+
+::
+
+   $ module purge
+
+This will not unload so-called `sticky modules
+<https://lmod.readthedocs.io/en/latest/240_sticky_modules.html>`__, which
+are special modules that do not normally need to be unloaded (for example
+because they define the appropriate module paths and possibly other environment
+variables). If really needed, sticky modules can be unloaded with
+``module --force purge``.
+
+
+Getting help
+~~~~~~~~~~~~
+
+To get a list of all available module commands, type:
+
+::
+
+   $ module help
+
+
+.. _collections of modules:
+
+Collections of modules
+~~~~~~~~~~~~~~~~~~~~~~
+
+Although it is convenient to set up your working environment by loading
+modules in your ``.bashrc`` or ``.profile`` file, this is error prone and
+you will end up shooting yourself in the foot at some point.
+
+The module system provides an alternative approach that lets you set up
+an environment with a single command, offering a viable alternative to
+polluting your ``.bashrc``.
+
+Define an environment
+
+   #. Be sure to start with a clean environment
+      ::
+
+         $ module purge
+
+   #. Load the modules you want in your environment, e.g.,
+      ::
+
+         $ module load matplotlib/3.7.2-gfbf-2023a
+         $ module load MATLAB/2023b
+
+   #. save your environment, e.g., as ``data_analysis``
+      ::
+
+          $ module save data_analysis
+
+Use an environment
+
+   ::
+
+      $ module restore data_analysis
+
+List all your environments
+
+   ::
+
+      $ module savelist
+
+Remove an environment
+
+   ::
+
+      $ rm ~/.lmod.d/data_analysis
+
+
+.. _specialized software stacks:
+
+Specialized software stacks
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The list of software available on a particular cluster can be
```

Comment:
The list of software available on a particular cluster as revealed by `module av` is overwhelmingly long.

Reply:
(addressed in a8164f746add7d7e5d8db59b0c06e0a5bfbe4356)